forge-game/src/main/java/forge/game/ability/effects/TextBoxExchangeEffect.java

```diff
@@ -0,0 +1,124 @@
+package forge.game.ability.effects;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Table.Cell;
+
+import forge.game.Game;
+import forge.game.ability.SpellAbilityEffect;
+import forge.game.card.Card;
+import forge.game.card.CardState;
+import forge.game.card.CardCopyService;
+import forge.game.event.GameEventCardStatsChanged;
+import forge.game.keyword.KeywordInterface;
+import forge.game.keyword.KeywordsChange;
+import forge.game.replacement.ReplacementEffect;
+import forge.game.spellability.SpellAbility;
+import forge.game.staticability.StaticAbility;
+import forge.game.trigger.Trigger;
+
+import java.util.List;
+
+/**
+ * Exchanges text boxes between two creatures.
+ */
+public class TextBoxExchangeEffect extends SpellAbilityEffect {
+    @Override
+    protected String getStackDescription(final SpellAbility sa) {
+        final List<Card> tgtCards = getTargetCards(sa);
+        Card c1;
+        Card c2;
+        if (tgtCards.size() == 1) {
+            c1 = sa.getHostCard();
+            c2 = tgtCards.get(0);
+        } else {
+            c1 = tgtCards.get(0);
+            c2 = tgtCards.get(1);
+        }
+        return c1 + " exchanges text box with " + c2 + ".";
+    }
+
+    @Override
+    public void resolve(final SpellAbility sa) {
+        final List<Card> tgtCards = getTargetCards(sa);
+        final Card host = sa.getHostCard();
+        if (tgtCards.isEmpty()) {
+            return;
+        }
+
+        final Card c1;
+        final Card c2;
+        if (tgtCards.size() == 1) {
+            c1 = host;
+            c2 = tgtCards.get(0);
+        } else {
+            c1 = tgtCards.get(0);
+            c2 = tgtCards.get(1);
+        }
+
+        // snapshot the original text boxes before modifying
+        final Card copy1 = CardCopyService.getLKICopy(c1);
```

Comment:
I suppose extra LKI gets you around some additional code but it would probably still be cleaner/safer if you'd just prepare the Arrays for each card and only after modify both